src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -530,6 +502,49 @@ object GraphFrame extends Serializable {
     GraphFrame(vv, ee)
   }
 
+  /**
+   * Given a GraphFrame and a GraphX graph derived from the GraphFrame,
+   * merge attributes from the GraphX graph into the original GraphFrame.
+   *
+   * This method is useful for doing computations using the GraphX API and then merging the results
+   * with a GraphFrame.  For example, given:
+   *  - GraphFrame `gf`
+   *  - GraphX Graph[String, Int] `gx` with a String vertex attribute we want to call "category"
```

Comment:
you should mention somewhere that the graphX object was created from the first graphframe gf, otherwise it sounds magical and will not work for non-integral ids.